### PR TITLE
enhanced add_electrical_series

### LINKED ISSUE #196: enhanced control over add_electrical_series write options
It was recently brought to our attention that the speed of NWB `write_recording` for `ecephys` data can be hampered in simple cases due to the extra add-ins we automatically use by default.

These changes will also be mirrored temporarily into spikeextractors, since those functions are ultimately the ones being used for this issue.

**Is your feature request related to a problem? Please describe.**
Allowing users full control over how electrical series are written in order to speed conversion, particularly for cases when data is small enough to fit in memory.

**Describe the solution you'd like**
a) Allow user to disable compression
b) Allow user to specify compression options like chunk size, level, etc.
c) Allow user to disable DataChunkIteration (in safe manner, check if there is space available to load full array into memory)

**Describe alternatives you've considered**
It may yet be possible to use `np.memmap` write functionality directly inside an NWBFile, but this may take some time to explore. It would certainly be the fastest method, however.

## Checklist

- [X] Have you ensured the feature or change was not already [reported](https://github.com/catalystneuro/nwb-conversion-tools/issues)?
- [X] Have you included a brief and descriptive title?
- [X] Have you included a clear description of the problem you are trying to solve?
- [X] Have you checked our [Contributing](https://nwb-conversion-tools.readthedocs.io/en/master/developer_guide.html) document?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
[fix #196](https://github.com/catalystneuro/nwb-conversion-tools/issues/196) 

## Motivation

Allowing user control over write options for `ecephys` electrical series in order to speed the process in certain cases.

## How to test the behavior?
New tests have been added for each option, as well as time reporting for write speed comparison.
```
!pytest
```

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Is your contribution compliant with our coding style? Read about [PEP8](https://www.python.org/dev/peps/pep-0008/) and [black](https://black.readthedocs.io/en/stable/the_black_code_style.html).
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?
- [X] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number?

### CODE REVIEW

File: nwb_conversion_tools/utils/spike_interface.py [hunk 1]
```diff
@@ -460,7 +460,8 @@ def add_electrical_series(
     use_times: bool = False,
     write_as: str = 'raw',
     es_key: str = None,
-    write_scaled: bool = False
+    write_scaled: bool = False,
+    use_compression: bool = True
```
Comment (bendichter) [inline nwb_conversion_tools/utils/spike_interface.py:464]: better to have compression='gzip', mirroring [h5py syntax](https://docs.h5py.org/en/stable/high/dataset.html#filter-pipeline)

Comment (CodyCBakerPhD) [inline nwb_conversion_tools/utils/spike_interface.py:464]:
```suggestion
    compression: Optional[str] = "gzip"
```
@bendichter Sure, how about this, with `None` meaning no compression?

**Comment by CodyCBakerPhD** [comment]
@bendichter This PR should be ready for review now.

I was thinking of adding some util functions that allow users to estimate NWB write speed on their given system with specified data types and write/parallelization options to help them understand how long it should take them to convert a given RecordingExtractor ahead of time, but that will all be in a subsequent PR.

**Comment by CodyCBakerPhD** [comment]
@bendichter This PR should be ready for review now.

I was thinking of adding some util functions that allow users to estimate NWB write speed on their given system with specified data types and write/parallelization options to help them understand how long it should take them to convert a given RecordingExtractor ahead of time, but that will all be in a subsequent PR.

**Comment by CodyCBakerPhD** [comment]
Granted, the changes we talked about with specification of Chunk size and enhancement of the underlying DataChunkIterator are not a part of this PR; I'm thinking a subsequent PR would be best given how technically focused that change will be.

**Comment by CodyCBakerPhD** [comment]
To-do: confirm compression carried through in tests of roundtrip nwbfiles

**Comment by CodyCBakerPhD** [comment]
@bendichter Added direct tests of final compression to confirm it went through as expected. Anything else to do here?

**Review by bendichter — APPROVED** [review]

**Pull request merged**